src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java

```diff
@@ -0,0 +1,234 @@
+import static uk.gov.hmcts.reform.managecase.client.datastore.ApprovalStatus.REJECTED;
+import static uk.gov.hmcts.reform.managecase.client.datastore.CaseDetails.APPROVAL_STATUS;
+import static uk.gov.hmcts.reform.managecase.client.datastore.CaseDetails.CASE_ROLE_ID;
+import static uk.gov.hmcts.reform.managecase.client.datastore.CaseDetails.ORGANISATION;
+import static uk.gov.hmcts.reform.managecase.client.datastore.CaseDetails.ORGANISATION_TO_ADD;
+import static uk.gov.hmcts.reform.managecase.client.datastore.CaseDetails.ORGANISATION_TO_REMOVE;
+
+@Service
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis", "PMD.PrematureDeclaration", "PMD.ExcessiveImports"})
+public class ApplyNoCDecisionService {
+
+    private final PrdRepository prdRepository;
+    private final DataStoreRepository dataStoreRepository;
+    private final NotifyService notifyService;
+    private final JacksonUtils jacksonUtils;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public ApplyNoCDecisionService(PrdRepository prdRepository,
+                                   DataStoreRepository dataStoreRepository,
+                                   NotifyService notifyService,
+                                   JacksonUtils jacksonUtils,
+                                   ObjectMapper objectMapper) {
+        this.prdRepository = prdRepository;
+        this.dataStoreRepository = dataStoreRepository;
+        this.notifyService = notifyService;
+        this.jacksonUtils = jacksonUtils;
+        this.objectMapper = objectMapper;
+    }
+
+    public Map<String, JsonNode> applyNoCDecision(ApplyNoCDecisionRequest applyNoCDecisionRequest) {
+        CaseDetails caseDetails = applyNoCDecisionRequest.getCaseDetails();
+        Map<String, JsonNode> data = caseDetails.getData();
+
+        if (data == null) {
+            throw new ValidationException(NO_DATA_PROVIDED);
+        }
+
+        // TODO: Use ACA-71 method when available on branch
+        JsonNode changeOrganisationRequestField = data.get("ChangeOrganisationRequestField");
+
+        String approvalStatus = getNonNullStringValue(changeOrganisationRequestField, APPROVAL_STATUS);
+        String caseRoleId = getNonNullStringValue(changeOrganisationRequestField, CASE_ROLE_ID);
+        validateCorFieldOrganisations(changeOrganisationRequestField);
+
+        if (NOT_CONSIDERED.getCode().equals(approvalStatus)) {
+            throw new ValidationException(NOC_REQUEST_NOT_CONSIDERED);
+        } else if (REJECTED.getCode().equals(approvalStatus)) {
+            nullifyNode(changeOrganisationRequestField);
+            return data;
+        } else if (!APPROVED.getCode().equals(approvalStatus)) {
+            throw new ValidationException(UNKNOWN_NOC_APPROVAL_STATUS);
+        }
+
+        applyDecision(caseDetails, changeOrganisationRequestField, caseRoleId);
+
+        nullifyNode(changeOrganisationRequestField);
```

Comment:
instead of nullifyNode, can't we remove this node?

Reply:
The design states we should nullify all the fields (otherwise agree with you!)